### ISSUE TITLE
IOS Safari year chosen is not year selected

### ISSUE THREAD

**Issue by NathanQ** [720kb/angular-datepicker#201]
When choosing a year in dropdown the selected year is always different by a different amount with each selection. The year does change, but to something else. If choosing the same year as selected, it will stay the same.

**Comment by tipycalFlow** [comment]
Looks like an issue with the `paginateYears` method. If I remove that function call from inside `setNewYear()`, it works (functionally, but with some minor css issues) for both web and mobile.

**Comment by tipycalFlow** [comment]
The issue seems to be that when the select DOM changes, the selected index remains the old one. I could fix it by changing `setNewYear` function: putting `j.year = Number(a), N(j.monthNumber, j.year)` in a `setTimeout()` with 100ms delay.

**Comment by tipycalFlow** [comment]
The issue seems to be that when the `select` DOM changes, the selected index doesn't change. @NathanQ, the issue can be reproduced on chrome for mac (inspect and toggle for mobile with Nexus device selected), so it's not an iOS issue but a datepicker issue. I could fix it by delaying setting `$scope.year` in `$scope.setNewYear` function (towards end):

```
$scope.paginateYears(year);
$scope.showYearsPagination = false;
setTimeout(function(){
   $scope.year = Number(year);
   setDaysInMonth($scope.monthNumber, $scope.year);
}, 100);
```

Check [pull request](https://github.com/720kb/angular-datepicker/compare/master...tipycalFlow:patch-1)

**Comment by NathanQ** [comment]
nice work @tipycalFlow

**Comment by 45kb** [comment]
@tipycalFlow thanks for this! does that work if you change `setTimeout()` with `$scope.$evalAsync()` ?

**Comment by tipycalFlow** [comment]
@45kb yup the timeout was only to pinpoint the issue (you can ignore the pull request), `$evalAsync()` is the angular way. I haven't tested with `$evalAsync()` though.

**Comment by 45kb** [comment]
@tipycalFlow great, if it works using evalAsync just PR it 👍

**Comment by tipycalFlow** [comment]
@45kb oddly, `$evalAsync` didn't work but `setTimeout()` works with 0 ms delay as well. Something's wrong with the digest cycle in our case, I guess.

**Comment by 45kb** [comment]
@45kb strange, evalAsync is same to $timeout(0) ... just use $timeout(0) and PR if you want 👍

**Comment by tipycalFlow** [comment]
@45kb tested with `$timeout(..., 0)` and [PR](https://github.com/720kb/angular-datepicker/pull/222) created

**Comment by 45kb** [comment]
@tipycalFlow thanks i am on it, can you reply there please? thanks a lot 👍

**Comment by roger2hk** [comment]
This issue can be closed as it has been delivered in 2.1.10.
